File: app/src/main/java/org/tasks/caldav/DeleteCalendarViewModel.java
package org.tasks.caldav;

import org.tasks.data.CaldavAccount;
import org.tasks.data.CaldavCalendar;
import org.tasks.ui.ActionViewModel;

@SuppressWarnings("WeakerAccess")
public class DeleteCalendarViewModel extends ActionViewModel {
  void deleteCalendar(CaldavClient client, CaldavAccount account, CaldavCalendar calendar) {
    run(() -> client.forCalendar(account, calendar).deleteCollection());
  }
}

File: app/src/main/java/org/tasks/caldav/DeleteCalendarViewModel.kt
package org.tasks.caldav

import dagger.hilt.android.lifecycle.HiltViewModel
import org.tasks.data.CaldavAccount
import org.tasks.data.CaldavCalendar
import org.tasks.ui.ActionViewModel
import javax.inject.Inject

@HiltViewModel
class DeleteCalendarViewModel @Inject constructor(
        private val provider: CaldavClientProvider
) : ActionViewModel() {
    suspend fun deleteCalendar(account: CaldavAccount, calendar: CaldavCalendar) {
        run {
            calendar.url?.let { provider.forAccount(account, it).deleteCollection() }
        }
    }
}